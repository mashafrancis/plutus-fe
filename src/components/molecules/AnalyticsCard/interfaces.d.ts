import { ReactNode } from 'react';

type color =
	| 'blueCard'
	| 'yellowCard'
	| 'purpleCard'
	| 'redCard'
	| 'greenCard'
	| 'brownCard';

export interface AnalyticsCardProps {
	icon?: ReactNode;
	mainInfo: string;
	subInfo: string | number | ReactNode;
	colorClass?: color;
	onClick?: () => void;
}
