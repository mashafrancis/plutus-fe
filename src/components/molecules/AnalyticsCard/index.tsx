import { AnalyticsCardProps } from './interfaces';
import { Box, ButtonBase, Grid, Typography } from '@mui/material';
import { useTheme } from '@mui/material/styles';

const AnalyticsCard = ({
	icon,
	mainInfo,
	subInfo,
	colorClass,
	onClick,
}: AnalyticsCardProps): JSX.Element => {
	const theme = useTheme();
	const classes = {
		root: {
			flexGrow: 1,
		},
		details: {
			display: 'flex',
			flexDirection: 'row',
		},
		content: {
			flex: '1 0 auto',
			flexDirection: 'row',
			width: '100%',
		},
		title: {
			fontWeight: 500,
		},
		cardPaper: {
			borderRadius: 12,
		},
		image: {
			width: 128,
			height: 128,
		},
		mainCard: {
			cursor: 'pointer',
			borderRadius: 2,
			height: 'fit-content',
			border: `1px solid ${theme.palette.divider}`,
			marginBottom: 1,
			[theme.breakpoints.up('sm')]: {
				marginRight: 1,
			},
			// [theme.breakpoints.down('sm')]: {
			// 	paddingTop: '0 !important',
			// 	marginBottom: 1,
			// },
		},
		mainInfo: {
			[theme.breakpoints.down('sm')]: {
				display: 'none',
			},
		},
		yellowCard: {
			color: '#71340F',
			backgroundColor: '#FEF2E8',
			borderColor: '#71340F',
		},

		blueCard: {
			color: '#1967d2',
			backgroundColor: '#e8f0fe',
			borderColor: '#1967d2',
		},

		purpleCard: {
			color: '#501174',
			backgroundColor: '#F4EEF8',
			borderColor: '#501174',
		},

		brownCard: {
			color: '#3e2723',
			backgroundColor: '#efebe9',
			borderColor: '#3e2723',
		},

		redCard: {
			color: '#34383B',
			backgroundColor: '#D5E4EB',
			borderColor: '#BFD7DF',
			// color: '#980910',
			// backgroundColor: '#F9E3E3',
			// borderColor: '#980910',
		},

		greenCard: {
			// color: '#0B6F66',
			// backgroundColor: '#E9F5E6',
			// borderColor: '#0B6F66',
			color: '#3E4E56',
			backgroundColor: '#D9E9BA',
			borderColor: '#3E4E56',
		},
	};

	return (
		<Grid item lg={4} md={6} xs={6}>
			<Box
				sx={{ ...classes[colorClass!], ...classes.mainCard }}
				onClick={onClick}
				data-testid="analytics-card"
			>
				<Box marginLeft={2} marginTop={1} display={{ sm: 'none' }}>
					<Typography
						classes={classes.mainInfo}
						fontWeight={500}
						fontSize={{ xs: 14, sm: 16 }}
						sx={{ ...classes[colorClass!] }}
						variant="h6"
						data-testid="main-info"
					>
						{mainInfo}
					</Typography>
				</Box>
				<Box
					classes={classes.content}
					padding={2}
					display={'flex'}
					alignItems={'center'}
				>
					<Box marginRight={2}>
						<ButtonBase>{icon}</ButtonBase>
					</Box>
					<Box
						display={'flex'}
						flexDirection={{ xs: 'column', sm: 'row' }}
						flex={'1 1 100%'}
						justifyContent={{ sm: 'space-between' }}
						alignItems={{ sm: 'center' }}
					>
						<Typography
							fontWeight={500}
							fontSize={{ xs: 12, sm: 14 }}
							sx={{ ...classes[colorClass!], ...classes.mainInfo }}
							variant="h6"
							data-testid="main-info"
						>
							{mainInfo}
						</Typography>
						<Typography
							variant="h4"
							fontWeight={600}
							sx={{ ...classes[colorClass!] }}
							data-testid="sub-info"
						>
							{subInfo}
						</Typography>
					</Box>
				</Box>
			</Box>
		</Grid>
	);
};

export default AnalyticsCard;
