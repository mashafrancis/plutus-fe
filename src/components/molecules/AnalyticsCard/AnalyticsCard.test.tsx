// react libraries
import { render, screen } from '@testing-library/react';

// component
import AnalyticsCard from './index';

describe('AnalyticsCard component', () => {
	const props = {
		mainInfo: 'mainInfo',
		subInfo: 'subInfo',
	};

	const { asFragment } = render(<AnalyticsCard {...props} />);

	it('should render correctly', () => {
		expect(asFragment()).toMatchSnapshot();

		const elem = screen.getByTestId('analytics-card');
		expect(elem).toHaveClass(
			'MuiPaper-root MuiCard-root makeStyles-root-1 makeStyles-cardPaper-5 MuiPaper-outlined MuiPaper-rounded',
		);

		const elemHeader = screen.getByTestId('main-info');
		expect(elemHeader.innerHTML).toBe('mainInfo');

		const elemDetails = screen.getByTestId('sub-info');
		expect(elemDetails.innerHTML).toBe('subInfo');
	});
});
