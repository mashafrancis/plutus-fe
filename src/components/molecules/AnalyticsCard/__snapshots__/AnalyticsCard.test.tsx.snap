// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnalyticsCard component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root makeStyles-mainCard-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-root-1 makeStyles-cardPaper-5 MuiPaper-outlined MuiPaper-rounded"
      data-testid="analytics-card"
      style="cursor: pointer;"
    >
      <div
        class="MuiCardContent-root makeStyles-content-3"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between MuiGrid-grid-xs-12"
          style="display: flex;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1 MuiGrid-grid-md-2"
          >
            <button
              class="MuiButtonBase-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-10 MuiGrid-grid-sm-true"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
              style="display: flex; width: 100%;"
            >
              <p
                class="MuiTypography-root makeStyles-mainInfoText-7 MuiTypography-body1"
                data-testid="main-info"
              >
                mainInfo
              </p>
              <h4
                class="MuiTypography-root makeStyles-subInfoText-8 MuiTypography-h4"
                data-testid="sub-info"
              >
                subInfo
              </h4>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
